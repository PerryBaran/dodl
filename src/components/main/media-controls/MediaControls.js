import React, { useContext } from 'react';
import style from './mediaControls.module.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPlay, faPause, faForward, faBackward } from '@fortawesome/free-solid-svg-icons';
import AppContext from '../../AppContext';

const MediaControls = (props) => {
    const { isPlaying, setIsPlaying, skipSong } = props
    const { hideWhilePlaying } = useContext(AppContext)

    return (
        <div className={`centerFlex positionTop`}>
                <div className={`${style.controls} ${style.middle}`}>
                    <button className={`${style.control} ${hideWhilePlaying(style.pause)}`} onClick={() => skipSong(false)}>
                        <FontAwesomeIcon icon={faBackward} />
                    </button>
                    <button className={`${style.control} ${hideWhilePlaying(style.pause)}`} onClick={() => setIsPlaying(!isPlaying)}>
                        <FontAwesomeIcon icon={isPlaying ? faPause : faPlay} />
                    </button>
                    <button className={`${style.control} ${hideWhilePlaying(style.pause)}`}>
                        <FontAwesomeIcon icon={faForward} onClick={() => skipSong()}/>
                    </button>
                </div>
            </div>
    );
};

export default MediaControls