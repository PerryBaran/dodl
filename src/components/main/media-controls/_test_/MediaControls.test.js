import React from "react";
import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom'
import MediaControls from "../MediaControls";
import AppContext from "../../../AppContext";
import renderer from "react-test-renderer";

afterEach(cleanup);

const RenderWithContext = (isPlaying = false) => {
    const hideWhilePlaying = (input) => {
        if (!isPlaying) {
          return input
        }
        return ''
    };
    return (
        <AppContext.Provider value={{hideWhilePlaying}}>
            <MediaControls/>
        </AppContext.Provider>
    )
};

it("renders without crashing", () => { 
    render(RenderWithContext());
});

it("contains 3 button elements", () => {
    render(RenderWithContext());
    const buttons = screen.getAllByRole('button');
    expect(buttons).toHaveLength(3);
});

it("shows className while paused", () => {
    render(RenderWithContext());
    const buttons = screen.getAllByRole('button');
    console.log(buttons)
    buttons.forEach((button) => {
        expect(button).toHaveClass('pause');
    })
});

it("hides className while playing", () => {
    render(RenderWithContext(true));
    const buttons = screen.getAllByRole('button');
    buttons.forEach((button) => {
        expect(button).not.toHaveClass('pause');
    })
});

it("matches snapshot while paused", () => {
    const tree = renderer.create(RenderWithContext()).toJSON();
    expect(tree).toMatchSnapshot();
});

it("matches snapshot while playing", () => {
    const tree = renderer.create(RenderWithContext(true)).toJSON();
    expect(tree).toMatchSnapshot();
});