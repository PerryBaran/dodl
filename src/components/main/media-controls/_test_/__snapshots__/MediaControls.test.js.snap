// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot while paused 1`] = `
<div
  className="centerFlex positionTop"
>
  <div
    className="controls middle"
  >
    <button
      className="control pause"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-backward "
        data-icon="backward"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M459.5 71.41l-171.5 142.9v83.45l171.5 142.9C480.1 457.7 512 443.3 512 415.1V96.03C512 68.66 480.1 54.28 459.5 71.41zM203.5 71.41L11.44 231.4c-15.25 12.87-15.25 36.37 0 49.24l192 159.1c20.63 17.12 52.51 2.749 52.51-24.62v-319.9C255.1 68.66 224.1 54.28 203.5 71.41z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <button
      className="control pause"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-play "
        data-icon="play"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M361 215C375.3 223.8 384 239.3 384 256C384 272.7 375.3 288.2 361 296.1L73.03 472.1C58.21 482 39.66 482.4 24.52 473.9C9.377 465.4 0 449.4 0 432V80C0 62.64 9.377 46.63 24.52 38.13C39.66 29.64 58.21 29.99 73.03 39.04L361 215z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <button
      className="control pause"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-forward "
        data-icon="forward"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M52.51 440.6l171.5-142.9V214.3L52.51 71.41C31.88 54.28 0 68.66 0 96.03v319.9C0 443.3 31.88 457.7 52.51 440.6zM308.5 440.6l192-159.1c15.25-12.87 15.25-36.37 0-49.24l-192-159.1c-20.63-17.12-52.51-2.749-52.51 24.62v319.9C256 443.3 287.9 457.7 308.5 440.6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`matches snapshot while playing 1`] = `
<div
  className="centerFlex positionTop"
>
  <div
    className="controls middle"
  >
    <button
      className="control "
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-backward "
        data-icon="backward"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M459.5 71.41l-171.5 142.9v83.45l171.5 142.9C480.1 457.7 512 443.3 512 415.1V96.03C512 68.66 480.1 54.28 459.5 71.41zM203.5 71.41L11.44 231.4c-15.25 12.87-15.25 36.37 0 49.24l192 159.1c20.63 17.12 52.51 2.749 52.51-24.62v-319.9C255.1 68.66 224.1 54.28 203.5 71.41z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <button
      className="control "
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-play "
        data-icon="play"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M361 215C375.3 223.8 384 239.3 384 256C384 272.7 375.3 288.2 361 296.1L73.03 472.1C58.21 482 39.66 482.4 24.52 473.9C9.377 465.4 0 449.4 0 432V80C0 62.64 9.377 46.63 24.52 38.13C39.66 29.64 58.21 29.99 73.03 39.04L361 215z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <button
      className="control "
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-forward "
        data-icon="forward"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M52.51 440.6l171.5-142.9V214.3L52.51 71.41C31.88 54.28 0 68.66 0 96.03v319.9C0 443.3 31.88 457.7 52.51 440.6zM308.5 440.6l192-159.1c15.25-12.87 15.25-36.37 0-49.24l-192-159.1c-20.63-17.12-52.51-2.749-52.51 24.62v319.9C256 443.3 287.9 457.7 308.5 440.6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </div>
</div>
`;
